### PR TITLE
Create intermediate sourceset to share between Apple and non-Apple targets

### PR DESCRIPTION
**Pull request by ankushg**
In this PR: 

- [x] create `nativeCoroutines` as a sourceset between `common` and `apple`
- [x] moved the contents of `apple*` into `nativeCoroutines*`
- [x] created a cross-platform `expect fun` for `freeze()`
- [x] created `PlatformError` as an `expect class`
	- exposed as `NSError` on Apple and `Throwable` on JS
	- create an internal `kotlinCause` extension that we that we can use instead of digging into `userInfo` so the tests work on non-Apple platforms

This sourceset can be used to house the Apple-agnostic code, and can eventually be used to share code across Apple and JS 😄 

## Notes:
- maybe it makes sense to keep the test cases in `appleTest` for now until we can successfully migrate to `kotlinx-coroutine-test`?

### CODE REVIEW

File: kmp-nativecoroutines-core/src/supportedTargetTest/kotlin/com/rickclephas/kmp/nativecoroutines/NativeFlowTests.kt [hunk 1]
```diff
@@ -1,29 +1,31 @@
 package com.rickclephas.kmp.nativecoroutines
```
Comment (rickclephas) [inline kmp-nativecoroutines-core/src/supportedTargetTest/kotlin/com/rickclephas/kmp/nativecoroutines/NativeFlowTests.kt:1]: These test classes can be reverted to the state from #44, right? 
I think moving them to the new "common" source set makes sense, we'll deal with the conversion to `coroutines-test` later 🙂

**Review by rickclephas — CHANGES_REQUESTED** [review]

File: kmp-nativecoroutines-core/src/appleMain/kotlin/com/rickclephas/kmp/nativecoroutines/NSError.kt [hunk 2]
```diff
@@ -6,6 +6,13 @@ import platform.Foundation.NSError
 import platform.Foundation.NSLocalizedDescriptionKey
 import kotlin.native.concurrent.freeze
 
+actual typealias PlatformError = NSError
+
+internal actual fun Throwable.asPlatformError(): PlatformError = this.asNSError()
+
+actual val PlatformError.kotlinCause
```
[the remark this answers is not in the document]
Comment (ankushg) [reply kmp-nativecoroutines-core/src/appleMain/kotlin/com/rickclephas/kmp/nativecoroutines/NSError.kt:13]: We really only use it in test cases to verify that we're throwing the correct exception

I'll move it to `nativeCoroutinesTest`

File: kmp-nativecoroutines-core/src/supportedTargetTest/kotlin/com/rickclephas/kmp/nativecoroutines/NativeFlowTests.kt (hunk 1 above)
Comment (ankushg) [reply kmp-nativecoroutines-core/src/supportedTargetTest/kotlin/com/rickclephas/kmp/nativecoroutines/NativeFlowTests.kt:1]: Going to rebase now that #44 has been merged 😄

File: kmp-nativecoroutines-core/src/appleMain/kotlin/com/rickclephas/kmp/nativecoroutines/NSError.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (rickclephas) [reply kmp-nativecoroutines-core/src/appleMain/kotlin/com/rickclephas/kmp/nativecoroutines/NSError.kt]: I actually think moving it to `nativeCoroutinesTest` makes the most sense. 
With #42 we'll get something similar to `asKotlinThrowable()` which will actually convert a `NativeError` to a `Throwable` (most likely by wrapping it). So then `kotlinCause` sounds great as a test utility.

(I might have to rethink these tests a little, `asPlatformError()` could be tested separately making the "common" tests a lot cleaner)

**Review by rickclephas — CHANGES_REQUESTED** [review]
Looks good, let's move `kotlinCause` to `nativeCoroutinesTest` and merge this 😄

**Review by rickclephas — APPROVED** [review]

**Pull request merged**